client/my-sites/plans-v2/plans-column/index.tsx

```diff
@@ -0,0 +1,131 @@
+import JetpackBundleCard from 'components/jetpack/card/jetpack-bundle-card';
+import JetpackPlanCard from 'components/jetpack/card/jetpack-plan-card';
+import FormattedHeader from 'components/formatted-header';
+
+/**
+ * Type dependencies
+ */
+import type { Duration, PurchaseCallback, ProductType, SelectorProduct } from '../types';
+
+interface PlanColumnType {
+	duration: Duration;
+	onPlanClick: PurchaseCallback;
+	productType: ProductType;
+	siteId: number | null;
+}
+
+type PlanWithBought = SelectorProduct & { owned: boolean; legacy: boolean };
+
+const PlanComponent = ( {
+	plan,
+	onClick,
+	currencyCode,
+}: {
+	plan: PlanWithBought;
+	onClick: PurchaseCallback;
+	currencyCode: string;
+} ) => {
+	const price =
+		useSelector( ( state ) => getProductCost( state, plan.costProductSlug || plan.productSlug ) ) ||
+		0;
+	const CardComponent = [ OPTIONS_JETPACK_SECURITY, OPTIONS_JETPACK_SECURITY_MONTHLY ].includes(
+		plan.productSlug
+	)
+		? JetpackBundleCard
+		: JetpackPlanCard;
+	return (
+		<CardComponent
+			iconSlug={ plan.iconSlug }
+			productName={ plan.displayName }
+			subheadline={ plan.tagline }
+			description={ plan.description }
+			currencyCode={ currencyCode }
+			billingTimeFrame={ durationToText( plan.term ) }
+			buttonLabel={ productButtonLabel( plan ) }
+			onButtonClick={ () => onClick( plan.productSlug ) }
+			features={ { items: [] } }
+			originalPrice={ price }
+			isOwned={ plan.owned }
+			deprecated={ plan.legacy }
+		/>
+	);
+};
+
+const PlansColumn = ( { duration, onPlanClick, productType, siteId }: PlanColumnType ) => {
+	const currencyCode = useSelector( ( state ) => getCurrentUserCurrencyCode( state ) );
+	const isFetchingProducts = useSelector( ( state ) => isProductsListFetching( state ) );
+	const currentPlan =
+		useSelector( ( state ) => getSitePlan( state, siteId ) )?.product_slug || null;
+
+	const planObjects: PlanWithBought[] = useMemo( () => {
+		let owned = false;
+		const plans: PlanWithBought[] = SELECTOR_PLANS.map( ( productSlug ) => {
+			const item = slugToItem( productSlug );
+			if ( ! owned && currentPlan ) {
+				owned = productSlug === currentPlan;
+			}
+			return item && itemToSelectorProduct( item );
+		} )
+			.filter(
+				( product: SelectorProduct | null ): product is SelectorProduct =>
+					!! product &&
+					duration === product.term &&
+					PRODUCTS_TYPES[ productType ].includes( product.productSlug )
+			)
+			.map( ( product: SelectorProduct ) => ( {
+				...product,
+				owned: product.productSlug === currentPlan,
+				legacy: false,
+			} ) );
+
+		if ( ! owned && currentPlan && currentPlan !== PLAN_JETPACK_FREE ) {
+			const item = slugToItem( currentPlan );
+			const currentPlanSelector = item && itemToSelectorProduct( item );
+			if ( currentPlanSelector ) {
+				plans.unshift( { ...currentPlanSelector, owned: true, legacy: true } );
+			}
+		}
+		return plans;
+	}, [ duration, productType, currentPlan ] );
```

Comment:
It took me sometime to understand what is going on here, I think it would nice to have a comment on top of the `useMemo` to explain the main gist.

Reply:
Added comments in 1ec6f0e1373c6edf8d338de1912dfb5de5b67123.